File: packages/ui/db/entity/Commentable.ts
import { cascadeOneToMany } from "@/db/cascadeOptions";
import BaseColumnSchemaPart from "@/db/entity/BaseColumnSchemaPart";
import { Comment } from "@/db/entity/Comment";
import { ICommentable } from "models";
import { EntitySchema } from "typeorm";

export type Commentable = ICommentable & {
  comments: Comment[];
};

const CommentableSchema = new EntitySchema<Commentable>({
  name: "commentable",
  columns: {
    ...BaseColumnSchemaPart,
    descriptor: {
      type: String,
    },
  },
  relations: {
    comments: {
      type: "one-to-many",
      target: "comment",
      inverseSide: "commentable",
      ...cascadeOneToMany,
    },
  },
});

export default CommentableSchema;

File: go-server/models/commentable.go
package models

type Commentable struct {
	ID         *int      `json:"id,omitzero"`
	Descriptor string    `json:"descriptor"`
	Comments   []Comment `json:"comments,omitzero"`
}

func (c Commentable) IsZero() bool {
	return c.ID == nil && c.Descriptor == "" && len(c.Comments) == 0
}
